### DISCUSSION TITLE
PolyData.area is not reliable?

### DISCUSSION THREAD

**Discussion by nimafo** [pyvista/pyvista#3877] [Q&A]
Hi, 
I have a cube, and I project it onto its bottom surface,
the .area attribute of the projected mesh gives twice the number as expected.
I am looking for the union area, but it seems that the overlapping area is calculated twice.
Any idea how can this be solved? 

Best,
N.

```
mesh = pv.Cube()
projected = mesh.project_points_to_plane()
projected.area, projected
```

output:
> (2.0,
>  PolyData (0x7faeb449ae80)
>    N Cells:	6
>    N Points:	8
>    N Strips:	0
>    X Bounds:	-5.000e-01, 5.000e-01
>    Y Bounds:	-5.000e-01, 5.000e-01
>    Z Bounds:	-8.660e-01, -8.660e-01
>    N Arrays:	3)

**Comment by darikg** [comment]
Consider checking the cell normals to select only the 'top' cells that face upwards:

```python
import pyvista as pv
import numpy as np

mesh = pv.Cube()
normal = np.array([0, 0, 1])
projected = mesh.project_points_to_plane(normal=normal)

direction = np.dot(projected.cell_normals, normal)  # vector of n_cell dot products. should be 1 for upward facing, -1 downward
upward_facing = projected.extract_cells(direction >= 0.999)  # Don't use == 1 in case of numerical error
upward_facing = pv.PolyData(upward_facing.points, faces=upward_facing.cells)  # convert from UnstructuredGrid
upward_facing.area

> 1.0
```

this isn't the same thing as the union area though -- for that I think you're going to need a 2d geometry library like shapely to convert all of the faces of the projected mesh to 2d polygons, then perform a union on them.
